src/coreclr/jit/emitriscv64.cpp

```diff
@@ -3659,6 +3684,50 @@ void emitter::emitDisInsName(code_t code, const BYTE* addr, instrDesc* id)
             }
             return;
         }
+        case 0b0101111: // AMO - atomic memory operation
+        {
+            unsigned int funct5 = code >> 27;
+            // clang-format off
+            const char* name =
+                funct5 == 0b00010 ? "lr" :
+                funct5 == 0b00011 ? "sc" :
+                funct5 == 0b00001 ? "amoswap" :
+                funct5 == 0b00000 ? "amoadd" :
+                funct5 == 0b00100 ? "amoxor" :
+                funct5 == 0b01100 ? "amoand" :
+                funct5 == 0b01000 ? "amoor" :
+                funct5 == 0b10000 ? "amomin" :
+                funct5 == 0b10100 ? "amomax" :
+                funct5 == 0b11000 ? "amominu" :
+                funct5 == 0b11100 ? "amomaxu" :
+                (assert(!"Illegal funct5 within atomic memory operation, emitDisInsName"), "?");
+            // clang-format on
+
+            unsigned int funct3 = (code >> 12) & 0x7;
+            // clang-format off
+            char width =
+                funct3 == 0b010 ? 'w' :
+                funct3 == 0b011 ? 'd' :
+                (assert(!"Illegal width tag within atomic memory operation, emitDisInsName"), '?');
+            // clang-format on
+
+            const char* aq = code & (1 << 25) ? "a" : "";
+            const char* rl = code & (1 << 26) ? "r" : "";
+
+            int len = printf("%s.%c.%s%s", name, width, aq, rl);
+            if (len <= 0)
+            {
+                return;
+            }
+            assert(len <= 12);
+
+            const char* dest = RegNames[(code >> 7) & 0x1f];
+            const char* addr = RegNames[(code >> 15) & 0x1f];
+            const char* data = RegNames[(code >> 20) & 0x1f];
+
+            printf("%*s %s, %s, %s\n", 12 - len, "", dest, addr, data);
```

Comment:
Please print risc-v instructions not yours. (*. https://github.com/llvm/llvm-project/blob/main/llvm/test/MC/RISCV/rv64a-valid.s)
If then, instruction size will exceed 12 length. Please update spaces in other instructions as well as atomic instructions.

Reply:
OK, I wanted to avoid changing padding everywhere, but I can do that and print `.aqrl` like LLVM.